workspaces/sandbox/plugins/sandbox/src/components/SandboxCatalog/__tests__/SandboxCatalogBanner.test.tsx

```diff
@@ -0,0 +1,258 @@
+
+const SandboxContext = React.createContext<{
+  loading: boolean;
+  userData?: SignupData;
+  verificationRequired: boolean;
+  pendingApproval: boolean;
+}>({
+  loading: false,
+  userData: undefined,
+  verificationRequired: false,
+  pendingApproval: false,
+});
+
+jest.mock(
+  '../../../assets/images/sandbox-banner-image.svg',
+  () => 'mocked-image-path',
+);
+jest.mock('../../../hooks/useSandboxContext', () => ({
+  useSandboxContext: jest.fn(),
+}));
+
+describe('SandboxCatalogBanner', () => {
+  const mockTheme = createTheme();
+  const mockUseSandboxContext = useSandboxContext as jest.MockedFunction<
+    typeof useSandboxContext
+  >;
+
+  beforeEach(() => {
+    mockUseSandboxContext.mockReturnValue({
+      loading: false,
+      userData: undefined,
+      userFound: false,
+      userReady: false,
+      verificationRequired: false,
+      pendingApproval: false,
+      fetchError: null,
+      signupError: null,
+      ansibleError: null,
+      ansibleData: undefined,
+      ansibleUIUser: undefined,
+      ansibleUIPassword: '',
+      ansibleUILink: undefined,
+      ansibleStatus: AnsibleStatus.NEW,
+      refetchUserData: jest.fn(),
+      signupUser: jest.fn(),
+      refetchAAP: jest.fn(),
+      userStatus: '',
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const renderWithProviders = (
+    contextValue: Partial<{
+      loading: boolean;
+      userData?: SignupData;
+      userFound: boolean;
+      userReady: boolean;
+      verificationRequired: boolean;
+      pendingApproval: boolean;
+      fetchError: string | null;
+      signupError: string | null;
+      ansibleError: string | null;
+      ansibleStatus: AnsibleStatus;
+    }> = {},
+  ) => {
+    mockUseSandboxContext.mockReturnValue({
+      loading: false,
+      userData: undefined,
+      userFound: false,
+      userReady: false,
+      verificationRequired: false,
+      pendingApproval: false,
+      fetchError: null,
+      signupError: null,
+      ansibleError: null,
+      ansibleStatus: AnsibleStatus.NEW,
+      refetchUserData: jest.fn(),
+      signupUser: jest.fn(),
+      refetchAAP: jest.fn(),
+      ansibleData: undefined,
+      ansibleUIUser: undefined,
+      ansibleUIPassword: '',
+      ansibleUILink: undefined,
+      userStatus: '',
+      ...contextValue,
+    });
+    return render(
+      <ThemeProvider theme={mockTheme}>
+        <SandboxContext.Provider
+          value={{
+            loading: false,
+            userData: undefined,
+            verificationRequired: false,
+            pendingApproval: false,
+            ...contextValue,
+          }}
+        >
+          <SandboxCatalogBanner />
+        </SandboxContext.Provider>
+      </ThemeProvider>,
+    );
+  };
+
+  it('renders loading skeleton when loading is true', () => {
+    renderWithProviders({ loading: true });
+    const skeletons = screen.getAllByTestId('MuiSkeleton-root');
+    expect(skeletons).toHaveLength(3);
+  });
```

Comment:
not sure I fully understand what is this testing 😕 .
I mean what are these skeletons , and why do we have 3 ?

Reply:
the 3 lines in the banner that shows up when the name and date is loading. 
Skeleton is a MUI component.